zebra/zebra_vty.c

```diff
@@ -357,9 +357,14 @@ static void show_nexthop_detail_helper(struct vty *vty,
 		break;
 	}
 
-	if ((re->vrf_id != nexthop->vrf_id)
-	    && (nexthop->type != NEXTHOP_TYPE_BLACKHOLE))
-		vty_out(vty, "(vrf %s)", vrf_id_to_name(nexthop->vrf_id));
+	if (re->vrf_id != nexthop->vrf_id) {
+		struct vrf *vrf = vrf_lookup_by_id(nexthop->vrf_id);
+
+		if (vrf)
+			vty_out(vty, "(vrf %s)", VRF_LOGNAME(vrf));
+		else
+			vty_out(vty, "(vrf UNKNOWN)");
```

Comment:
VRF_LOGNAME returns Uknown already, no need for this else output.

Reply:
Ok, will fix it.